packages/types/src/utils/executionAddress.ts

```diff
@@ -0,0 +1,48 @@
+import {keccak256} from "ethereum-cryptography/keccak.js";
+import {ByteVectorType} from "@chainsafe/ssz";
+
+export type ByteVector = Uint8Array;
+
+export class ExecutionAddressType extends ByteVectorType {
+  constructor() {
+    super(20, {typeName: "ExecutionAddress"});
+  }
+  toJson(value: ByteVector): unknown {
+    const str = super.toJson(value) as string;
+    return toChecksumAddress(str);
+  }
+}
+
+function isAddressValid(address: string): boolean {
+  return /^(0x)?[0-9a-f]{40}$/i.test(address);
```

Comment:
this regex would fail if you pass mixed case address or upper case but those are valid addresses

Reply:
The trailing `i` is [ignoreCase](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/RegExp/ignoreCase)

Reply:
ah right, overlooked that